hoomd/hpmc/update.py

```diff
@@ -512,6 +513,207 @@ def N(self):  # noqa: N802 - allow N as a function name
         return N_dict
 
 
+class Shape(Updater):
+    """Apply shape updates to the shape definitions defined in the integrator.
+
+    Args:
+        trigger (Trigger): Call the updater on triggered time steps.
+
+        step_size (float): Default maximum size of shape trial moves.
+
+        pretend (bool, optional): When True the updater will not actually update
+            the shape definitions. Instead, moves will be proposed and the
+            acceptance statistics will be updated correctly.
+
+        nselect (int, optional): Number of types to change every time the
+            updater is called.
+
+        nsweeps (int, optional): Number of times to update shape definitions
+            during each triggered timesteps.
+
+        multi_phase (bool, optional): When True MPI is enforced and shapes are
+            updated together for two boxes.
+
+        num_phase (int, optional): How many boxes are simulated at the same
+            time, supported values are 1, 2 and 3.
+
+    Examples::
+
+        mc = hoomd.hpmc.integrate.ConvexPolyhedron(23456)
+        mc.shape["A"] = dict(vertices=numpy.asarray([(1, 1, 1), (-1, -1, 1),
+                                                    (1, -1, -1), (-1, 1, -1)]) / 2)
+        vertex_move = hoomd.hpmc.shape_move.Vertex(stepsize={'A': 0.01},
+                                                   param_ratio=0.2,
+                                                   volume=1.0)
+        updater = hoomd.hpmc.update.Shape(shape_move=vertex_move,
+                                          trigger=hoomd.trigger.Periodic(1),
+                                          nselect=1,
+                                          nsweeps=1,
+                                          multi_phase=False,
+                                          num_phase=1)
+
+    Attributes:
+        trigger (Trigger): Call the updater on triggered time steps.
+
+        step_size (`TypeParameter` [``particle type``, `float`]): Maximum size
+            of shape trial moves.
```

Comment:
Why is this a property of the `Shape` updater rather than the chosen move?

Reply:
That's how it was originally implemented but I changed the ownership of `step_size` to the Updater to make the shape moves simpler and more specific. I don't have a strong opinion either way. The only question I have is which of these approaches will play better with the current tuning infrastructure? There is not a shape `step_size` tuner yet but I will implement one in a separate PR.

Reply:
Either will work fine with a tuner. It might be cleaner if the same object that owns step size also provides the move acceptance counters - otherwise the user would need to pass both objects to the tuner.

Reply:
Currently, both the step size and the acceptance counter are owned by the updater.